torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1445,7 +1445,8 @@ Example::
                 >>> ddp_model._egister_comm_hook(state = None, hook = allreduce)
 
             .. warning ::
-                ``get_future`` API supports only NCCL backend.
+                ``get_future`` API supports NCCL, and partially GLOO and MPI backends
+                (support is limited to non peer-to-peer operations).
```

Comment:
Feel it's better to explicitly mention the operators here with some examples.

You can reword it as "no support for peer-to-peer operations like send/recv"